itest/rust/src/derive_variant.rs

```diff
@@ -0,0 +1,82 @@
+use crate::itest;
+use godot::bind::ToVariant;
+use godot::prelude::*;
+
+#[derive(ToVariant)]
+struct Struct1;
+
+#[itest(focus)]
+fn unit_struct() {
+    let v = Struct1.to_variant();
+    assert_eq!(v, dict! {"Struct1": Variant::nil()}.to_variant());
+}
+
+#[derive(ToVariant)]
+struct Struct2(i32, i32);
+
+#[itest(focus)]
+fn tuple_struct() {
+    let s = Struct2(1, 2);
+
+    let v = s.to_variant();
+    assert_eq!(v, dict! {"Struct2": array![1,2]}.to_variant());
+}
+
+#[derive(ToVariant)]
+struct Struct3 {
+    field1: i32,
+    field2: i32,
+}
+
+#[itest(focus)]
+fn named_struct() {
+    let s = Struct3 {
+        field1: 4,
+        field2: 5,
+    };
+    let v = s.to_variant();
+    assert_eq!(
+        v,
+        dict! {"Struct3": dict! {"field1" : 4, "field2": 5}}.to_variant()
+    );
+}
+
+#[derive(ToVariant)]
+struct Struct4(i32);
+#[itest(focus)]
```

Comment:
Please add an empty line before new definitions.

Also, don't use `focus` for code to be merged, it disables all other tests.
(We validate that in CI already, so you cannot accidentally do that).